src/Kovaleva.HomeWork-4/Task.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+
+namespace Kovaleva.HomeWork_4
+{
+    public class Task
+    {
+        private readonly string Id;
+
+        private string GetId()
+        { return Guid.NewGuid().ToString().ToUpper().Substring(0, 5); }
+
+        public string Name { get; set; }
+        public string Description { get; set; }
+
+        public DateTime StartDate { get; set; }
+
+        public DateTime EndDate { get; set; }
+
+        public Task()
+        {
+            Id = GetId();
+
+        }
+
+        public Task(string name, string description, DateTime startDate, DateTime endDate)
+        {
+            Id = GetId();
```

Comment:
this can be called by parameterless constructor like that (that is more OOP way)

public Task(string name, string description, DateTime startDate, DateTime endDate) : this()

Reply:
Да, это возможности языка С#. См. https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-constructors